lib/chef/provider/package/openbsd.rb

```diff
@@ -37,25 +38,42 @@ class Openbsd < Chef::Provider::Package
 
         def initialize(*args)
           super
-          @current_resource = Chef::Resource::Package.new(@new_resource.name)
-          @new_resource.source(pkg_path) if !@new_resource.source
+          @current_resource = Chef::Resource::Package.new(new_resource.name)
         end
 
         def load_current_resource
-          @current_resource.package_name(@new_resource.package_name)
+          @current_resource.package_name(new_resource.package_name)
           @current_resource.version(installed_version)
           @current_resource
         end
 
+        def define_resource_requirements
+          super
+
+          # Below are incomplete/missing features for this package provider
+          requirements.assert(:all_actions) do |a|
+            a.assertion { !new_resource.source }
+            a.failure_message(Chef::Exceptions::Package, 'The openbsd package provider does not support the source attribute')
+          end
+          requirements.assert(:all_actions) do |a|
+            a.assertion do
+              if new_resource.package_name =~ /^(.+?)--(.+)/
+                !new_resource.version
+              else
+                true
+              end
+            end
+            a.failure_message(Chef::Exceptions::Package, 'The openbsd package provider does not support providing a version and flavor')
```

Comment:
Because it never worked. The person that wrote the code didn't account for
this case. I'm just making it explicitly clear so you don't get some crazy
exception

On Fri, Mar 20, 2015, 6:57 PM Phil Dibowitz <notifications@github.com>
wrote:

> In lib/chef/provider/package/openbsd.rb
> <https://github.com/chef/chef/pull/3109#discussion_r26887591>:
>
> > +          super
> > +
> > +          # Below are incomplete/missing features for this package provider
> > +          requirements.assert(:all_actions) do |a|
> > +            a.assertion { !new_resource.source }
> > +            a.failure_message(Chef::Exceptions::Package, 'The openbsd package provider does not support the source attribute')
> > +          end
> > +          requirements.assert(:all_actions) do |a|
> > +            a.assertion do
> > +              if new_resource.package_name =~ /^(.+?)--(.+)/
> > +                !new_resource.version
> > +              else
> > +                true
> > +              end
> > +            end
> > +            a.failure_message(Chef::Exceptions::Package, 'The openbsd package provider does not support providing a version and flavor')
>
> I'm not sure I understand this. why doesn't this support vim-x11 1.2.4?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/chef/chef/pull/3109/files#r26887591>.
>